### DISCUSSION TITLE
Is there a way to print an iFrame instead of the top level document?

### DISCUSSION THREAD

**Discussion by Grandpappy** [MicrosoftEdge/WebView2Feedback#4307] [Q&A] [answered]
In our application, we are hosting some older html content, and in them it has an iframe with more content than will fit on the screen.  So it uses a scrollbar.  I would like to invoke PrintToPdf on the iframe's content, and not the top level document that contains the iframe.

When window.print() is called from within the iframe, it prints correctly where the printout will contain multiple pages to contain all the info in the iframe.  When PrintToPdf is called, it's giving the same output as window.print() in the iframe's containing page, which is the top level document, and the resulting print is a single page with the iframe's content showing up with a scrollbar in the printout.

So, is there a way to invoke PrintToPdf in the same way that window.print would be done for the iframe's document, and not the top level document?

Examples:
Printing at the top level document causing the iframe content to have a scrollbar 
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/4137294/f05dd661-40b2-4016-b251-6399b52e5244)

When printing from within the iframe (notice multiple pages) - this is the output I'm trying to get to:
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/4137294/856d1194-bcd2-449d-b15d-fcd5463f7ac1)

**Comment by victorhuangwq** [comment]
@monica-ch since this is related to the printtopdf function, could you take a look at it?

**Comment by monica-ch** [comment, marked as answer]
@Grandpappy Our Print APIs only apply to the WebView object, not any frames.

We are tracking a few similar requests in https://github.com/MicrosoftEdge/WebView2Feedback/issues/1990, https://github.com/MicrosoftEdge/WebView2Feedback/issues/2949, and https://github.com/MicrosoftEdge/WebView2Feedback/issues/3563.
